core/src/main/java/hudson/model/TopLevelItemDescriptor.java

```diff
@@ -114,6 +127,64 @@ public String getDisplayName() {
     }
 
     /**
+     * A description of this kind of item type. This description can contain HTML code but it is recommend to use text plain
+     * in order to avoid how it should be represented.
+     *
+     * @return A string, an empty string by default.
+     *
+     * @since TODO
+     */
+    @Nonnull
+    public String getDescription() {
+        try {
+            WebApp webapp = WebApp.getCurrent();
+            MetaClass meta = webapp.getMetaClass(this);
+            Script s = meta.loadTearOff(JellyClassTearOff.class).findScript("newInstanceDetail");
+            if (s == null) {
+                return "";
+            }
+            DefaultScriptInvoker dsi = new DefaultScriptInvoker();
+            StringWriter sw = new StringWriter();
+            XMLOutput xml = dsi.createXMLOutput(sw, true);
+            dsi.invokeScript(Stapler.getCurrentRequest(), Stapler.getCurrentResponse(), s, this, xml);
+            return sw.toString();
+        } catch (Exception e) {
+            return "";
+        }
+    }
+
+    /**
+     * Used to categorize this kind of item type. @see {@link ItemCategory}
+     *
+     * @return A string with the category identifier, {@link ItemCategory.UncategorizedCategory#getId()} by default.
+     *
+     * @since TODO
+     */
+    @Nonnull
+    public String getCategoryId() {
+        return new ItemCategory.UncategorizedCategory().getId();
+    }
+
+    /**
+     * @since TODO
```

Comment:
Consider that a :bug: since you could never guess at the magic `:size` template field. You _could_ guess (though it would be better to be told) what values the size might take, or that after expanding this field, the result should be an image resource path relative to the context root.